### PR TITLE
feat(stats): Expose discarded outcome

### LINKED ISSUE #59320: Expand client outcomes on the stats page
Today, we only expose three categories to our users.

Screenshot 2023-11-02 at 22 39 38

This makes debugging issues with the SDKs too cumbersome.
We should expose some stats we show on `_admin`.

Screenshot 2023-11-02 at 22 41 05

### PR DESCRIPTION
**Pull request by priscilawebdev**
### Goal

Expose the discarded outcome on the stats page, so users can better debug.

**Before:**
image

**After:**
image

closes https://github.com/getsentry/sentry/issues/59320

### CODE REVIEW

File: static/app/views/organizationStats/usageStatsOrg.tsx [hunk 1]
```diff
@@ -427,6 +440,24 @@ class UsageStatsOrganization<
             return;
           }
 
+          if (outcome === Outcome.CLIENT_DISCARD) {
+            usageStats[i].discarded += stat;
```
Comment (vgrozdanic) [inline static/app/views/organizationStats/usageStatsOrg.tsx:444]: Nit, but due to consistency with other part of code inside the function maybe we could also use: `usageStats[i][outcome] += stat;` here.

On the next refactor Friday, we can refactor this function then, cause it seems to me that it can be written more concise, but that is outside of the scope of this PR :)

Comment (priscilawebdev) [reply static/app/views/organizationStats/usageStatsOrg.tsx:444]: you have good eyes :) but we cannot do that, because we get 'client_discarded' from the backend and in the frontend we are naming it 'discarded'

Comment (priscilawebdev) [reply static/app/views/organizationStats/usageStatsOrg.tsx:444]: the function can for sure be improved but as you said it is outside of the scope of this PR  - maybe in a follow-up

Comment (vgrozdanic) [reply static/app/views/organizationStats/usageStatsOrg.tsx:444]: My bad, I missed that, makes sense now why are we using it like this, thanks :)

**Review by vgrozdanic — APPROVED** [review]

**Review by matejminar — APPROVED** [review]
I'd prefer this card to be last.
Accepted -> Filtered -> Dropped -> Discarded
Basically descending order of what's happening.

**Comment by priscilawebdev** [comment]
closing this PR in favor of https://github.com/getsentry/sentry/pull/73869
